### PR TITLE
Implemented real-time updates for private chats

### PR DESCRIPTION
**Pull request by Nath-Assa**
### Key changes:

- Implemented `observeMessages(user1: String, user2: String)` in both Mock and Firebase message repositories to enable real-time updates in chat interactions.
-  Developed a private `observeMessages()` function within `ChatViewModel` to encapsulate the logic for observing chat messages. This function is called in the init block.
- Wrote comprehensive tests for all functions to ensure they handle both success and failure scenarios effectively.
- Conducted a live application testing to validate the functionality and real-time performance of the messaging feature.

### CODE REVIEW

File: app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt:76]: It is unconventional to have all of this logic in the `update` block. To separate the act of updating the UI state and getting the user ID, this logic should be outside of the `update` block. If there is a particular reason that you are doing this please let me know (and leave a comment in the code if applicable)

**Review by jgeaso1266 — COMMENTED** [review]
One comment. Also, you put live app testing in your TODOs in the description. That should be done before merging.

**Comment by Nath-Assa** [comment]
> One comment. Also, you put live app testing in your TODOs in the description. That should be done before merging.

Yes, this is planned before merging.

File: app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt
Comment (Nath-Assa) [reply app/src/main/java/com/github/se/eventradar/viewmodel/ChatViewModel.kt:76]: Do you mean init block?

File: app/src/test/java/com/github/se/eventradar/ChatViewModelUnitTest.kt
Comment (christinehahaa) [inline app/src/test/java/com/github/se/eventradar/ChatViewModelUnitTest.kt:317]: These will likely need to be updated after the message repo update, will communicate offline for help with this

**Comment by Nath-Assa** [comment]
**Live Application Testing**:
The listener seems to work perfectly. In the Firestore Database, I added a message to a message history's message list and updated the message history's latest message id. The changes were immediately reflected in the app without any manual intervention. 
image
image
image

**Comment by Nath-Assa** [comment]
Live Application Testing pt2:
@jgeaso1266 and I both ran the app simultaneously and sent each other private messages. The app automatically updated the chats and behaved as expected.

**Review by christinehahaa — APPROVED** [review]
Nice job fixing bugs and testing. CI just needs to pass, then LGTM

**Comment by Nath-Assa** [comment]
After the fixes, I have conducted another final live app test and the live updates still work perfectly.

**Pull request merged**
